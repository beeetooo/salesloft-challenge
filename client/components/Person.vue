<template>
  <div class="person">
    <div class="person__circle">
      <p class="person__first-letter">
        {{ name[0] }}
      </p>
    </div>
    <div class="person__meta">
      <span class="person__name"> {{ name }} </span>
      <span class="person__title"> {{ title }} </span>
      <span class="person__email"> {{ email }} </span>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Person',

  props: {
    name: {
      type: String,
      required: true
    },
    title: {
      type: String,
      required: true,
    },
    email: {
      type: String,
      required: true,
    }
  },
}
</script>

<style lang="scss">
.person {
  padding: 12px 0 12px 0;
  background-color: #FFFFFF;
  border-radius: 4px;

  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;

  &__circle {
    width: 54px;
    height: 54px;
    border-radius: 50%;
    margin-bottom: 20px;

    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;

    background-color: #18577B;
  }

  &__meta {
    width: 100%;
    display: block;
  }

  &__name {
    display: block;
    color: black;
  }

  &__title {
    display: block;
    font-style: italic;
  }

  &__email {
    display: block;
    text-overflow: ellipsis;
  }
}
</style>
