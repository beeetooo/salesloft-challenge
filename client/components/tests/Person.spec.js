import { shallowMount } from '@vue/test-utils';

import Person from '../Person.vue';
import PersonPageObject from './Person.po';

describe('Person', () => {
  const mountComponent = () => {
    const wrapper = shallowMount(Person, {
      propsData: {
        name: 'Juan Pablo',
        title: 'Software Engineer',
        email: 'juanpablo@gmail.com'
      }
    });

    return new PersonPageObject(wrapper);
  };

  it('should render a person correctly', () => {
    const person = mountComponent();

    expect(person.name).toEqual('Juan Pablo');
    expect(person.title).toEqual('Software Engineer');
    expect(person.email).toEqual('juanpablo@gmail.com');
  });

  it('should match a snapshot', () => {
    expect(
      shallowMount(Person, {
        propsData: {
          name: 'Juan Pablo',
          title: 'Software Engineer',
          email: 'juanpablo@gmail.com'
        }
      })
    ).toMatchSnapshot();
  });
})
