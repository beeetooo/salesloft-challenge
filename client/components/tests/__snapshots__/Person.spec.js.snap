// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should match a snapshot 1`] = `
<div class="person">
  <div class="person__circle">
    <p class="person__first-letter">
      J
    </p>
  </div>
  <div class="person__meta"><span class="person__name"> Juan Pablo </span> <span class="person__title"> Software Engineer </span> <span class="person__email"> juanpablo@gmail.com </span></div>
</div>
`;
